File: test/nodejs/hydro_duck_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`hydro_duck extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.HYDRO_DUCK_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
    });

    it('hydro_duck function should return expected string', function (done) {
        db.all("SELECT hydro_duck('Sam') as value;", function (err, res) {
            if (err) throw err;
            assert.deepEqual(res, [{value: "HydroDuck Sam 🐥"}]);
            done();
        });
    });

    it('hydro_duck_openssl_version function should return expected string', function (done) {
        db.all("SELECT hydro_duck_openssl_version('Michael') as value;", function (err, res) {
            if (err) throw err;
            assert(res[0].value.startsWith('HydroDuck Michael, my linked OpenSSL version is OpenSSL'));
            done();
        });
    });
});

File: test/python/hydro_duck_test.py
import duckdb
import os
import pytest

# Get a fresh connection to DuckDB with the hydro_duck extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('HYDRO_DUCK_EXTENSION_BINARY_PATH')
    if (extension_binary == ''):
        raise Exception('Please make sure the `HYDRO_DUCK_EXTENSION_BINARY_PATH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    return conn

def test_hydro_duck(duckdb_conn):
    duckdb_conn.execute("SELECT hydro_duck('Sam') as value;");
    res = duckdb_conn.fetchall()
    assert(res[0][0] == "HydroDuck Sam 🐥");

def test_hydro_duck_openssl_version_test(duckdb_conn):
    duckdb_conn.execute("SELECT hydro_duck_openssl_version('Michael');");
    res = duckdb_conn.fetchall()
    assert(res[0][0][0:51] == "HydroDuck Michael, my linked OpenSSL version is OpenSSL");